adapter/src/dummy.rs

```diff
@@ -34,24 +34,29 @@ impl Adapter for DummyAdapter {
         Ok(())
     }
 
-    fn whoami(&self) -> String {
+    fn whoami(&self) -> ValidatorId {
         self.identity.clone()
     }
 
     fn sign(&self, state_root: &str) -> AdapterResult<String> {
         let signature = format!(
             "Dummy adapter signature for {} by {}",
             state_root,
-            self.whoami()
+            self.whoami().to_string()
         );
         Ok(signature)
     }
 
-    fn verify(&self, signer: &str, _state_root: &str, signature: &str) -> AdapterResult<bool> {
+    fn verify(
+        &self,
+        signer: &ValidatorId,
+        _state_root: &str,
+        signature: &str,
+    ) -> AdapterResult<bool> {
         // select the `identity` and compare it to the signer
         // for empty string this will return array with 1 element - an empty string `[""]`
         let is_same = match signature.rsplit(' ').take(1).next() {
-            Some(from) => from == signer,
+            Some(from) => from == signer.to_string(),
```

Comment:
or `Some(ref from) => from == signer`

Reply:
`signer` is a `ValidatorId`, `to_string` converts it to the string representation, so this won't work